### ISSUE TITLE
sort the prefixes by popularity

### ISSUE THREAD

**Issue by VladimirAlexiev** [nichtich/RDF-NS#17]
Run this several times 

```
perl -MRDF::NS -e "print RDF::NS->new(1)->PREFIX('http://www.w3.org/2001/XMLSchema#')"
```

and you get different answers, eg `xs xds` etc.

The doc says "If multiple prefixes are defined, it is not determinstic which one is returned!" But I wonder why it has to be so:

If you ask http://prefix.cc about `http://www.w3.org/2001/XMLSchema#` it confidently answers `xsd`. It has these voting buttons so worse alternatives like `xs xds` are not returned. Can't the module use this voting info?

**Comment by VladimirAlexiev** [comment]
http://prefix.cc/popular/all.txt seems to return them in the right order:
* for http://www.w3.org/2001/XMLSchema#: xsd, xs, xds, xmls (the last two are wrong)
* crm is the official namespace, ecrm is erlangen, etc.

**Comment by kjetilk** [comment]
I suppose it comes down to an implementation detail, that the mapping is represented internally as a hash, and therefore unordered. So, it is probably just a matter of having time to change it so that there is a an array of prefixes... I tend to agree, this behaviour is rather odd, but someone has to set aside time to do it. :-)

**Comment by nichtich** [comment]
The current release returns them in sorted order - I don't mind changing the order to something else. This only affects reverse lookup which is less relevant because every RDF serialization should declare its prefix-namespace mapping anyway.

**Comment by VladimirAlexiev** [comment]
Version 20150725 returned `xmls`.
Version 20170111 returns `xds`. I guess by "sorted order" you mean "alphabetically sorted": but that's not good, it should be "sorted by popularity" (i.e. as in http://prefix.cc/popular/all.txt)

> every RDF serialization should declare its prefix-namespace mapping

If your module does reverse lookup in `prefix.cc`, please do it right. Else remove this function.

**Comment by kjetilk** [comment]
Yeah, I agree that sorted by popularity would be a better solution, but I suppose that's a different issue?

**Comment by VladimirAlexiev** [comment]
The problem is that the reverse lookup returns BAD prefixes (eg `xds`) for known namespaces (eg `xsd`).
Now the issue title corresponds to what we're discussing. Please reopen it.

**Comment by nichtich** [comment]
This feature requires an extension of [the internal database](https://github.com/nichtich/RDF-NS/blob/master/share/prefix.cc) and [the update mechanism](https://github.com/nichtich/RDF-NS/blob/master/lib/RDF/NS.pm#L208) for reverse lookup. Each prefix could get an additional popularity ranking (1,2...n) among all prefixes that share the same namespace (additional column in `prefix.cc`). Updates will not only cover changes of prefix-namespace-mappings but also changes in popularity. The main work is in updating the UPDATE method but this is not on the top of my TODO list, so pull requests are welcome.

By the way [RDF::SN uses a different sort order](https://github.com/nichtich/RDF-NS/blob/master/lib/RDF/SN.pm#L22) to prefer shorter prefixes.
